server/issue.go

```diff
@@ -209,6 +211,72 @@ func httpAPICreateIssue(ji Instance, w http.ResponseWriter, r *http.Request) (in
 	return http.StatusOK, nil
 }
 
+func httpWorkflowCreateIssue(ji Instance, w http.ResponseWriter, r *http.Request) (int, error) {
+	if r.Method != http.MethodPost {
+		return http.StatusMethodNotAllowed,
+			errors.New("method " + r.Method + " is not allowed, must be POST")
+	}
+
+	var activationParams workflowclient.ActionActivationParams
+	if err := json.NewDecoder(r.Body).Decode(&activationParams); err != nil {
+		return http.StatusBadRequest,
+			errors.WithMessage(err, "failed to decode incoming request")
+	}
+
+	create := &struct {
+		UserId string           `json:"user_id"`
+		Fields jira.IssueFields `json:"fields"`
+	}{}
+	if err := json.NewDecoder(bytes.NewReader(activationParams.Action)).Decode(&create); err != nil {
+		return http.StatusBadRequest,
+			errors.WithMessage(err, "failed to decode incoming request specific parameters")
+	}
+
+	var client interface {
+		CreateIssue(issue *jira.Issue) (*jira.Issue, error)
+	}
+	if create.UserId != "" {
+		jiraUser, err := ji.GetPlugin().userStore.LoadJIRAUser(ji, create.UserId)
+		if err != nil {
+			return http.StatusInternalServerError, err
+		}
+
+		client, err = ji.GetClient(jiraUser)
+		if err != nil {
+			return http.StatusInternalServerError, err
+		}
+	} else {
+		jci, ok := ji.(*jiraCloudInstance)
+		if !ok {
+			return http.StatusBadRequest, errors.New("UserId is required for jira server instances.")
+		}
+
+		jiraClient, err := jci.getJIRAClientForServer()
+		if err != nil {
+			return http.StatusInternalServerError, fmt.Errorf("unable to get jira client for server: %w", err)
+		}
+
+		client = &JiraClient{Jira: jiraClient}
+	}
+
+	issue := &jira.Issue{
+		Fields: &create.Fields,
+	}
+
+	created, err := client.CreateIssue(issue)
+	if err != nil {
+		return http.StatusInternalServerError, errors.Errorf("Failed to create issue. %s", err.Error())
+	}
+
+	outputVars := make(workflowclient.Vars)
+	outputVars["TicketKey"] = created.Key
```

Comment:
Stupid question, but are the outputVars defined anywhere, or are we relying on documentation to communicate what will be returned?

Reply:
There is facility for expressing them on registration, but we don't currently do anything with that. (I imagined it for the UI). So yes, we are entirely reliant on docs for this.